platform/os_windows.go

```diff
@@ -189,6 +206,170 @@ func SetSdnRemoteArpMacAddress() error {
 	return nil
 }
 
+func HasMellanoxAdapater() bool {
+	adapterName, err := getMellanoxAdapterName()
+	if err != nil {
+		log.Errorf("Error while getting mellanox adapter name: %v", err)
+		return false
+	} else if adapterName == "" {
+		log.Printf("No Mellanox adapter found")
+		return false
+	}
+	log.Printf("Name of Mellanox adapter : %v", adapterName)
+	return true
+}
+
+// Regularly monitors the Mellanox PriorityVLANGTag registry value and sets it to desired value if needed
+func MonitorAndSetMellanoxRegKeyPriorityVLANTag() {
+	for {
+		adapterName, err := getMellanoxAdapterName()
+		if err == nil && adapterName != "" {
+			err := SetMellanoxPriorityVLANTag(adapterName)
+			if err != nil {
+				log.Errorf("error while monitoring and setting Mellanox Reg Key value: %v", err)
+			}
+			time.Sleep(mellanoxPriorityVLANTagMonitorInterval)
```

Comment:
better to use a `time.Ticker` than sleeping in a loop

Reply:
sure let me check